cosmos/_triggers/watcher.py

```diff
@@ -0,0 +1,133 @@
+            },
+        )
+
+    async def get_xcom_val_af3(self, key: str) -> Any | None:
+        from airflow.sdk.execution_time.xcom import XCom
+
+        return await sync_to_async(XCom.get_one)(
+            run_id=self.run_id,
+            key=key,
+            task_id=self.producer_task_id,
+            dag_id=self.dag_id,
+            map_index=self.map_index,
+        )
+
+    async def get_xcom_val_af2(self, key: str) -> Any | None:
+        from airflow.models import TaskInstance
+        from airflow.utils.session import create_session
+
+        def _get_xcom_val() -> Any | None:
+            with create_session() as session:
+                ti = (
+                    session.query(TaskInstance)
+                    .filter_by(
+                        dag_id=self.dag_id,
+                        task_id=self.producer_task_id,
+                        run_id=self.run_id,
+                    )
+                    .one()
+                )
+                return ti.xcom_pull(task_ids=self.producer_task_id, key=key)
+
+        return await sync_to_async(_get_xcom_val)()
+
+    async def get_xcom_val(self, key: str) -> Any | None:
+        if AIRFLOW_VERSION < Version("3.0.0"):
+            return await self.get_xcom_val_af2(key)
+        else:
+            return await self.get_xcom_val_af3(key)
+
+    async def _parse_node_status(self) -> str | None:
+        key = f"nodefinished_{self.model_unique_id.replace('.', '__')}" if self.use_event else "run_results"
+
+        compressed_xcom_val = await self.get_xcom_val(key)
+        if not compressed_xcom_val:
+            return None
+
+        data_json = _parse_compressed_xcom(compressed_xcom_val)
+
+        if self.use_event:
+            return data_json.get("data", {}).get("run_result", {}).get("status")  # type: ignore[no-any-return]
+
+        results = data_json.get("results", [])
+        node_result: dict[str, Any] = next(
+            (r for r in results if r.get("unique_id") == self.model_unique_id),
+            {},
+        )
+        return node_result.get("status")
+
+    async def run(self) -> AsyncIterator[TriggerEvent]:
+        self.log.info("Starting WatcherTrigger for model: %s", self.model_unique_id)
+
+        while True:
+            dr_state = await self.get_xcom_val("state")
+            node_status = await self._parse_node_status()
+            if node_status == "success":
+                self.log.info("Model '%s' succeeded", self.model_unique_id)
+                yield TriggerEvent({"status": "success"})  # type: ignore[no-untyped-call]
+                return
+            elif node_status == "failed":
+                self.log.warning("Model '%s' failed", self.model_unique_id)
+                yield TriggerEvent({"status": "failed"})  # type: ignore[no-untyped-call]
+                return
+            elif dr_state == "failed":
+                yield TriggerEvent({"status": "failed"})  # type: ignore[no-untyped-call]
```

Comment:
can we make the messaging consistent with poke method here to include in the error logs

"The dbt build command failed in producer task. Please check the log of task {self.producer_task_id} for details."